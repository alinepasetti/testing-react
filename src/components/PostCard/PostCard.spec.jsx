import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { PostCard } from ".";

const MOCK_PROPS = {
  title: "title 1",
  id: 1,
  body: "banana",
  image: {
    url: "url.com",
    title: "image title",
  },
};

describe("<PostCard />", () => {
  it("displays the title correctly", () => {
    render(<PostCard post={MOCK_PROPS} />);
    const title = screen.getByRole("heading", { name: "1 title 1" });
    expect(title).toBeInTheDocument();
  });

  it("displays the content correctly", () => {
    render(<PostCard post={MOCK_PROPS} />);
    const image = screen.getByRole("img", { name: "image title" });
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute("src", "url.com");
  });

  it("displays the image correctly", () => {
    render(<PostCard post={MOCK_PROPS} />);
    const content = screen.getByText("banana");
    expect(content).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const postCardElement = renderer
      .create(<PostCard post={MOCK_PROPS} />)
      .toJSON();
    expect(postCardElement).toMatchSnapshot();
  });
});
