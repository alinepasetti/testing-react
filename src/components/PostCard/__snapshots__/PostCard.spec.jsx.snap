// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  className="post"
>
  <img
    alt="image title"
    className="post-image"
    src="url.com"
  />
  <div
    className="post-content"
  >
    <h2>
      1
       
      title 1
    </h2>
    <p>
      banana
    </p>
  </div>
</div>
`;
